src/components/AssetSelection/components/AssetChainDropdown/AssetChainDropdown.tsx

```diff
@@ -0,0 +1,89 @@
+import type { ButtonProps, MenuProps } from '@chakra-ui/react'
+import {
+  Button,
+  Menu,
+  MenuButton,
+  MenuItemOption,
+  MenuList,
+  MenuOptionGroup,
+  Tooltip,
+} from '@chakra-ui/react'
+import type { AssetId } from '@shapeshiftoss/caip'
+import { useCallback, useMemo } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { isAssetSupportedByWallet } from 'state/slices/portfolioSlice/utils'
+import {
+  selectAssetChainNameById,
+  selectRelatedAssetIdsInclusiveSorted,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { AssetRowLoading } from '../AssetRowLoading'
+import { AssetChainRow } from './AssetChainRow'
+
+type AssetChainDropdownProps = {
+  assetId?: AssetId
+  onClick: (arg: AssetId | undefined) => void
+  buttonProps?: ButtonProps
+  isLoading?: boolean
+  isError?: boolean
+} & Omit<MenuProps, 'children'>
+
+export const AssetChainDropdown: React.FC<AssetChainDropdownProps> = ({
+  assetId,
+  onClick,
+  buttonProps,
+  isLoading,
+  isError,
+  ...menuProps
+}) => {
+  const {
+    state: { wallet },
+  } = useWallet()
+  const translate = useTranslate()
+  const assetChainName = useAppSelector(state => selectAssetChainNameById(state, assetId ?? ''))
+  const relatedAssetIds = useAppSelector(state =>
+    selectRelatedAssetIdsInclusiveSorted(state, assetId ?? ''),
+  )
+
+  const renderChains = useMemo(() => {
+    return relatedAssetIds.map(assetId => {
+      const isSupported = wallet && isAssetSupportedByWallet(assetId, wallet)
+
+      return (
+        <MenuItemOption value={assetId} key={assetId} isDisabled={!isSupported}>
+          <AssetChainRow assetId={assetId} />
+        </MenuItemOption>
+      )
+    })
+  }, [relatedAssetIds, wallet])
+
+  const onChange = useCallback((value: string | string[]) => onClick(value as AssetId), [onClick])
+
+  const isDisabled = useMemo(() => {
+    return relatedAssetIds.length <= 1 || isLoading || isError
+  }, [relatedAssetIds, isError, isLoading])
+
+  const buttonTooltipText = useMemo(() => {
+    if (relatedAssetIds.length > 1 || isLoading || isError) return ''
+    return translate('trade.tooltip.noRelatedAssets', { chainName: assetChainName })
+  }, [assetChainName, isError, isLoading, relatedAssetIds.length, translate])
+
+  if (!assetId || isLoading) return <AssetRowLoading {...buttonProps} />
+
+  return (
+    <Menu {...menuProps}>
+      <Tooltip label={buttonTooltipText}>
```

Comment:
see https://github.com/shapeshift/web/pull/6176/files#r1490732318

```suggestion
      <Tooltip isDisabled={isDisabled} label={buttonTooltipText}>
```

Reply:
🧠

Reply:
logic there is inverted, we need to render the tooltip when the menu is disabled, but a great suggestion nevertheless
dfd16b20bb085e3e0207274ac77fb15bd9cf0df0

Reply:
dfd16b20bb085e3e0207274ac77fb15bd9cf0df0

Reply:
was more nuanced than that, ce89008f017549252db31b1a6b688e6178c3f2f7